### DISCUSSION TITLE
Request Triangle Arbitrage Multicycle Example

### DISCUSSION THREAD

**Discussion by zhihuba** [rodrigo-brito/ninjabot#215] [Q&A] [answered]
It's a good project and I checked the documentation. Can be used to make portfolios, and custom cycles. But how to implement a strategy similar to the triangle arbitrage. or handle both 1-hour and 4-hour candles in one strategy

**Comment by rodrigo-brito** [comment, marked as answer]
Hi @zhihuba, you can use a stretegy in timeframe of `1h` and resample it to `4h` with the `GroupCandles` of `talib`:
https://github.com/markcheno/go-talib/blob/master/talib.go#L5949

**Comment by adamantmm** [comment]
`GroupCandles` is fine when the higher TF is an exact multiple and you only need closed bars. For triangle arb the harder part is not resampling — it is keeping three books coherent under the same clock.

If you go that route I would:

- treat each leg’s mid/ask/bid as its own feed, and only evaluate the cycle when all three have a quote newer than your max-staleness budget
- size from the bottleneck leg after fees, not from the pretty mid cross
- paper it with intentional 100–300ms lag on one leg; a lot of “free” triangles disappear once one book is slightly stale

Multi-TF trend strategies and triangle cycles share the feed plumbing but almost none of the risk logic — worth keeping as separate strategy types.
